src/Checkout_Blocks/js/postnl-fill-in-with/block.js

```diff
@@ -0,0 +1,110 @@
+import { useEffect, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { getSetting } from '@woocommerce/settings';
+import { useDispatch } from '@wordpress/data';
+import { store as noticesStore } from '@wordpress/notices';
+
+export const FillBlock = ( { checkoutExtensionData } ) => {
+    const postnlData = getSetting( 'postnl-for-woocommerce-blocks_data', {} );
+    const [ showButton, setShowButton ] = useState( false );
+    const [ isLoading, setIsLoading ]   = useState(false);
+    const { setBillingAddress, setShippingAddress } = useDispatch('wc/store/cart');
+    const { createErrorNotice } = useDispatch( noticesStore );
+
+	useEffect( () => {
+		if ( postnlSettings?.is_enabled_for_checkout ) {
+			setShowButton( true );
+		}
+	}, [ postnlData ] );
+
+	const prefillCheckoutFields = async () => {
+		try {
+			const response = await fetch( postnlSettings.ajaxUrl, {
+				method: 'POST',
+				headers: {
+					'Content-Type': 'application/x-www-form-urlencoded',
+				},
+				body: new URLSearchParams( {
+					action: 'get_postnl_user_info',
+					nonce: postnlSettings.ajaxNonce,
+				} ),
+			} );
```

Comment:
@OlfatProgressus , do you know if we can use `extensionCartUpdate()` then we can extend the WC store API to get the data from the backend?